### PR TITLE
Add API call for deleting withdraw address

### PR DESCRIPTION
**Pull request by ec**

### CODE REVIEW

File: app/api/api_v2/withdraws.rb [hunk 1]
```diff
@@ -51,6 +51,20 @@ class Withdraws < Grape::API
       present order, with: APIv2::Entities::WithdrawAddress
     end
 
+    desc 'Delete withdraw address.', scopes: %w[ withdraw ]
+    params do
+      requires :id, type: Integer, desc: "Withdraw address ID"
+    end
+    delete '/withdraws/addresses' do
```
Comment (yivo) [inline app/api/api_v2/withdraws.rb]: @mafarain URL should respect http://jsonapi.org/

File: app/api/api_v2/withdraws.rb [hunk 2]
```diff
@@ -51,6 +51,20 @@ class Withdraws < Grape::API
       present order, with: APIv2::Entities::WithdrawAddress
     end
 
+    desc 'Delete withdraw address.', scopes: %w[ withdraw ]
+    params do
+      requires :id, type: Integer, desc: "Withdraw address ID"
```
Comment (yivo) [inline app/api/api_v2/withdraws.rb]: @mafarain We use single quotes.

Comment (yivo) [reply app/api/api_v2/withdraws.rb]: And we don't mix single and double quotes.

File: app/api/api_v2/withdraws.rb [hunk 3]
```diff
@@ -51,6 +51,20 @@ class Withdraws < Grape::API
...
+    end
+    delete '/withdraws/addresses' do
+      withdraw_address = FundSource.find(params[:id])
+
+      if withdraw_address.delete
```
Comment (yivo) [inline app/api/api_v2/withdraws.rb]: @mafarain You should use destroy.

File: app/api/api_v2/withdraws.rb [hunk 4]
```diff
@@ -51,6 +51,20 @@ class Withdraws < Grape::API
...
+        status 200
+      else
+        status 500
```
Comment (yivo) [inline app/api/api_v2/withdraws.rb]: @mafarain LOL. This is not related to internal server error. Use 422 in such cases.

**Review by yivo — CHANGES_REQUESTED** [review]
@mafarain Please add specs for this. Examples: https://github.com/rubykube/peatio/pull/410 https://github.com/rubykube/peatio/pull/398

File: app/api/api_v2/withdraws.rb (hunk 4 above)
Comment (ec) [reply app/api/api_v2/withdraws.rb]: Just thought 422 is not really the error that fit. Okay, will change though

File: app/api/api_v2/withdraws.rb (hunk 2 above)
Comment (ec) [reply app/api/api_v2/withdraws.rb]: Sorry, my fault. BTW, there is some mixes that comes not from my patch. Will change that also

File: app/api/api_v2/withdraws.rb [hunk 5]
```diff
@@ -51,6 +51,20 @@ class Withdraws < Grape::API
       present order, with: APIv2::Entities::WithdrawAddress
     end
 
+    desc 'Delete withdraw address.', scopes: %w[ withdraw ]
+    params do
+      requires :id, type: Integer, desc: 'Withdraw address ID'
+    end
+    delete '/withdraws/addresses' do
```
Comment (yivo) [inline app/api/api_v2/withdraws.rb]: @mafarain Can you please add ID parameter here and we will merge? I think we will defer specs for this since we   have small amount of time.

**Review by yivo — CHANGES_REQUESTED** [review]

**Review by yivo — APPROVED** [review]
